### PR TITLE
Struct groups

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by tpoterba** [comment]
The Table ones are kinda broken because they get reordered because the `Struct` constructor hash-orders the fields. This means that `df.row.dtype != df.schema`, which is bad. We can fix this by moving to python 3

File: python/hail/api2/table.py [hunk 1]
```diff
@@ -1967,5 +1967,33 @@ def explode(self, field):
 
         return Table(self._jt.explode(fields_rev[field]))
 
+    @property
+    @handle_py4j
+    def globals(self):
+        """Returns a struct expression including all global fields.
```
Comment (catoverdrive) [inline python/hail/api2/table.py]: add a test for tables also?

Comment (tpoterba) [reply python/hail/api2/table.py]: will remove the table ones like we talked about

**Comment by cseed** [comment]
Github branch protection for breaking was too strong.  Fixed.

**Pull request merged**